### PR TITLE
update users_guide: add links to PyAMG and PETSc documentation

### PR DESCRIPTION
**Pull request by vlukes**

### CODE REVIEW

File: doc/links.inc [hunk 1]
```diff
@@ -99,3 +99,6 @@
 .. _openscad: http://www.openscad.org
 .. _python scripting in freecad: https://www.freecadweb.org/wiki/Python_scripting_tutorial
 .. _netgen: https://ngsolve.org
+
+.. _PyAMG Tutorial: https://github.com/pyamg/pyamg/wiki/Tutorial
+.. _PETSc Users Manual: http://www.mcs.anl.gov/petsc/petsc-current/docs/manual.pdf
```
Comment (rc) [inline doc/links.inc]: Or provide link to the html docs as well? It might be more practical.

File: doc/users_guide.rst [hunk 2]
```diff
@@ -1392,9 +1392,10 @@ available:
   solver for all examples. It is strongly recommended to install umfpack
   and its SciPy wrappers to get good performance.
 - 'ls.scipy_iterative': Interface to SciPy iterative solvers.
-- 'ls.pyamg': Interface to PyAMG solvers.
+- 'ls.pyamg': Interface to PyAMG solvers, see `PyAMG Tutorial`_.
 - 'ls.petsc': Interface to Krylov subspace solvers of PETSc supporting parallel
-  use.
+  use. See `PETSc Users Manual`_, Chapter 4, for more details about available
+  Krylov methods and preconditioners.
```
Comment (rc) [inline doc/users_guide.rst]: What about linking the relevant example(s) here? Not sure - it is a complex topic.

**Comment by rc** [comment]
Thanks! Better docs for non-trivial linear solver use are sorely needed. There is just `examples/multi_physics/biot_short_syntax.py` and the parallel examples now.

**Comment by vlukes** [comment]
Unfortunately, there are no suitable html pages. The pdf file in the 
link is the best I have found. You can look at pages [1], section 
"Linear Solvers (KSP)", but the pdf has more readable overview of the 
available methods.

[1] http://www.mcs.anl.gov/petsc/petsc-current/docs/

**Comment by rc** [comment]
Besides the comments above:

- PyAMG: add also links to:
  - https://github.com/pyamg/pyamg/wiki
  - and especially https://github.com/pyamg/pyamg/wiki/Examples

**Comment by rc** [comment]
True, but https://www.mcs.anl.gov/petsc/documentation/index.html should be listed as it is the main entry point.

**Comment by rc** [comment]
One could link also to http://www.mcs.anl.gov/petsc/petsc-current/src/ksp/ksp/examples/tutorials/ etc. but it is not directly usable from python (though petsc4py maintains the same interface pretty well).

**Pull request merged**